third_party/amd/include/Dialect/TritonAMDGPU/IR/TritonAMDGPUOps.td

```diff
@@ -152,6 +152,22 @@ def InstructionSchedHint : TT_AMDGPU_Op<"instruction_sched_hint", []> {
   let assemblyFormat = [{ attr-dict }];
 }
 
+def condBarrierOp :
```

Comment:
Can we extract the op definition and its associated lowering out as a separate pull request? Also along the way please add lit test for the lowering procedure.

Reply:
Sure, that sounds better.